examples/docker/echo_app.py

```diff
@@ -0,0 +1,52 @@
+# Example using SkyPilot python API to run the echo app in a container.
+#
+# The echo example ingests a file, prints the contents and writes it back out.
+# In this YAML, the output is mapped to a Sky Storage object, which writes to a
+# cloud bucket.
+#
+# Usage:
+#   python echo_app.py
+
+import random
+import sky
+import string
+
+with sky.Dag() as dag:
+    # The setup command to build the container image
+    setup = 'docker build -t echo:v0 /echo_app'
+
+    # The command to run - runs the container and mounts volumes
+    run = 'docker run --rm ' \
+          '--volume="/inputs:/inputs:ro" ' \
+          '--volume="/outputs:/outputs:rw" ' \
+          'echo:v0 /inputs/README.md /outputs/output.txt'
+
+    echo_app = sky.Task(
+        setup=setup,
+        run=run,
+    )
+
+    # Configure file mounts to copy local contents to remote
+    echo_app.set_file_mounts({
+        '/inputs': './echo_app',
+        '/echo_app': './echo_app',
+    })
+
+    # Configure outputs for the task - we'll write to a bucket using Sky Storage
+    output_bucket_name = ''.join(random.choices(string.ascii_lowercase, k=15))
+    output_storage = sky.Storage(name=output_bucket_name)
```

Comment:
Should we add mount mode here?

Reply:
Rebased on #1223 to support specifying storage mode here - will ship this only after that's merged